Source/SwiftLintFramework/Rules/Idiomatic/StaticOperatorRule.swift

```diff
@@ -79,23 +78,34 @@ public struct StaticOperatorRule: ASTRule, ConfigurationProviderRule, OptInRule
         ]
     )
 
-    public func validate(file: SwiftLintFile, kind: SwiftDeclarationKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        guard kind == .functionFree,
-            let offset = dictionary.offset,
-            let name = dictionary.name?.split(separator: "(").first.flatMap(String.init) else {
-                return []
-        }
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
 
-        let characterSet = CharacterSet(charactersIn: name)
-        guard characterSet.isDisjoint(with: .alphanumerics) else {
-            return []
+private extension StaticOperatorRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
+
+        override func visitPost(_ node: FunctionDeclSyntax) {
+            if node.isFreeFunction, node.isOperator {
+                violationPositions.append(node.funcKeyword.positionAfterSkippingLeadingTrivia)
+            }
         }
+    }
+}
 
-        return [
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, byteOffset: offset))
-        ]
+private extension FunctionDeclSyntax {
+    var isFreeFunction: Bool {
+        parent?.is(CodeBlockItemSyntax.self) ?? false
+    }
+
+    var isOperator: Bool {
+        switch identifier.tokenKind {
+        case .spacedBinaryOperator:
+            return true
+        default:
+            return false
+        }
```

Comment:
```suggestion
        identifier.tokenKind == .spacedBinaryOperator
```

Reply:
so this enum actually has an associated value, so we can't use an equality comparison. We could use `if case let` but I personally don't love that syntax.